### PR TITLE
Simplify backupCodec setting.

### PR DESCRIPTION
**Pull request by cloudwebrtc**

### CODE REVIEW

File: lib/src/options.dart [hunk 1]
```diff
@@ -200,16 +201,19 @@ class VideoPublishOptions {
...
   const VideoPublishOptions({
-    this.videoCodec = 'H264',
+    this.videoCodec = defaultVideoCodec,
     this.videoEncoding,
     this.simulcast = true,
     this.videoSimulcastLayers = const [],
     this.screenShareSimulcastLayers = const [],
     this.name,
-    this.backupCodec,
+    this.enableBackupCodec = false,
```
Comment (lukasIO) [inline lib/src/options.dart:215]: I think we want to enable it by default

**Review by lukasIO — CHANGES_REQUESTED** [review]

File: lib/src/options.dart [hunk 2]
```diff
@@ -219,7 +223,8 @@ class VideoPublishOptions {
     List<VideoParameters>? videoSimulcastLayers,
     List<VideoParameters>? screenShareSimulcastLayers,
     String? videoCodec,
-    BackupVideoCodec? backupCodec,
+    bool? enableBackupCodec,
```
Comment (davidzhao) [inline lib/src/options.dart:226]: I think we ended up settling for having this inside `BackupVideoCodec` type?

Comment (cloudwebrtc) [reply lib/src/options.dart]: Removed `bool? enableBackupCodec`, only use `nullable backupVideoCodec`

**Comment by davidzhao** [comment]
how would someone explicitly disable this?

**Comment by cloudwebrtc** [comment]
okay, updated the BackupVideoCodec structure,   
It will be enabled when backupVideoCodec != null and backupVideoCodec.enabled == true. Do we need to set default for backupCodec? Enabled by default unless the user wants to disable

**Comment by lukasIO** [comment]
yeah, by default it should be enabled unless e2ee is enabled (which automatically needs to disable backup codecs)

**Review by davidzhao — APPROVED** [review]
lgtm

**Pull request merged**